### PR TITLE
Expose tlsNegotiationTimeout for netty client

### PR DESCRIPTION
**Pull request by zoewangg**
## Motivation and Context
This is to utilize the tlsNegotiationTimeout from the future `sdk-default-configuration.json`
Apache HTTP client 4.x and URLConnection currently don't support handshake timeout

## Description
Expose tlsNegotiationTimeout for netty client to allow users configure it.

## Testing
Added unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [x] My change requires a change to the Javadoc documentation
- [x] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpConfigurationOption.java [hunk 1]
```diff
@@ -117,12 +117,20 @@
     public static final SdkHttpConfigurationOption<TlsTrustManagersProvider> TLS_TRUST_MANAGERS_PROVIDER =
         new SdkHttpConfigurationOption<>("TlsTrustManagersProvider", TlsTrustManagersProvider.class);
 
+    /**
+     * An upper limit on how long from the time a TLS handshake is allowed to take from the time the CLIENT HELLO message is
+     * sent to the time the client and server have fully negotiated ciphers and exchanged keys.
```
Comment (cenedhryn) [inline http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpConfigurationOption.java:122]: I can't understand exactly what the text means, but maybe it's the `from the time` on line 121 that should just be removed?

**Review by cenedhryn — COMMENTED** [review]
What's the current default value of the TLS negotiation timeout?

File: http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpConfigurationOption.java (hunk 1 above)
Comment (zoewangg) [reply http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpConfigurationOption.java:122]: Yeah, copy paste error, will fix

**Comment by zoewangg** [comment]
The current default value of the TLS negotiation timeout is 10s in netty and that's why I kept it 10s here as well

**Review by cenedhryn — APPROVED** [review]

**Pull request merged**